src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -508,7 +508,7 @@ def register_modules(self, **kwargs):
             setattr(self, name, module)
 
     def __setattr__(self, name: str, value: Any):
-        if hasattr(self, name) and hasattr(self.config, name):
+        if name in self.__dict__ and hasattr(self.config, name):
```

Comment:
Yes essentially it's to prevent throwing a deprecation warning in the very beginning when the __init__ of StableDiffusionPipeline is called. `hasattr(...)` would work but it'd throw a warning when the config has `unet` but the pipeline has not yet defined the `self.unet`

Reply:
Oh right, sure. Thanks for clarifying!